modules/formulize/include/entriesdisplay.php

```diff
@@ -4578,10 +4577,14 @@ function formulize_gatherDataSet($settings=array(), $searches, $sort="", $order=
             $dataHandler = new formulizeDataHandler(false);
             $metadataFieldTypes = $dataHandler->metaDataFieldTypes;
 
-            foreach (array_keys($metadataFieldTypes) as $field) {
-                print $field;
-                print $metadataFieldTypes[$field];
-                $ele_type = $metadataFieldTypes[$field];
+            if (isset($metadataFieldTypes[$key])){
```

Comment:
Well, you don't want to look through the entire array... look at the code there, you will end up with $ele_type equal to the type of the last field in the array. You want to simply set $ele_type to the type of the $key... $ele_type = $metadataFieldTypes[$key];